mkosi

```diff
@@ -1600,56 +1595,57 @@ def invoke_tdnf(args: CommandLineArguments,
     with mount_api_vfs(args, root):
         run(cmdline, check=True)
 
-@completestep('Installing Photon')
-def install_photon(args: CommandLineArguments, root: str, do_run_build_script: bool) -> None:
-    masked = disable_kernel_install(args, root)
-
-    gpg_key = '/etc/pki/rpm-gpg/VMWARE-RPM-GPG-KEY'
-    gpg_key_string = f'file://{gpg_key}'
 
-    if os.path.exists(gpg_key):
-        gpgcheck = "gpgcheck=1"
-        cmdline = ["rpm", "--import", gpg_key, "--root", root]
-        run(cmdline, check=True)
+def setup_dnf(args: CommandLineArguments, root: str,
+              repos: List[Tuple[str, str, str, str, Optional[str]]] = []) -> None:
```

Comment:
This type signature is pretty much inscrutable. :) It would be better if this were a dataclass or (since we're still on 3.6) a named tuple.

Reply:
I actually had no idea about NamedTuple, neat!

Reply:
Yes, they are great, but have been mostly supplanted by dataclasses on 3.7+, since they are still tuples, i.e. they are immutable (generally a good thing) and compare with tuples of the same size (might not be wanted, if they are supposed to represent different types), but for now they are the best thing we have.